runtime/runtime.go

```diff
@@ -927,10 +931,14 @@ func (r *interpreterRuntime) emitAccountEvent(
 		))
 	}
 
+	var err error
 	exportedEvent := exportEvent(eventValue)
 	wrapPanic(func() {
-		runtimeInterface.EmitEvent(exportedEvent)
+		err = runtimeInterface.EmitEvent(exportedEvent)
 	})
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Is this the right way to revert the transaction?

Reply:
@ramtinms yep, that looks good!